pydantic/json_schema.py

```diff
@@ -732,10 +732,26 @@ def literal_schema(self, schema: core_schema.LiteralSchema) -> JsonSchemaValue:
         # jsonify the expected values
         expected = [to_jsonable_python(v) for v in expected]
 
-        result: dict[str, Any] = {'enum': expected}
+        result: dict[str, Any] = {}
         if len(expected) == 1:
             result['const'] = expected[0]
 
+        if self._config.json_schema_literal_type == 'enum':
+            result['enum'] = expected
+        elif self._config.json_schema_literal_type == 'oneof-const':
+            # TODO (reesehyde): do we want this condition or not? why do we still produce 'enum' for single values?
+            if len(expected) > 1:
```

Comment:
It seems intentional that the `"enum": [<value>]` tag gets added as well as `"const": <value>` but I'm not quite sure why. I assume that we wouldn't want:

```json
{
    "const": "SingleValue",
    "oneOf": [{"const": "SingleValue"}]
}
```
over just the `"const": "SingleValue"`, but since I don't follow why we add the `enum` in the single-value case I wanted to check.

Reply:
Per [this comment](https://github.com/pydantic/pydantic/issues/8888#issuecomment-2005499567)

> we recently re-added enum even when it is a single item (where we used const alone previously) because the orval API client generator mishandled the case where const was specified and enum was not. (And I was not aware of any backwards incompatibility that that might have been introducing.)